src/stream_base.cc

```diff
@@ -452,6 +452,8 @@ void StreamBase::AddMethods(Environment* env, Local<FunctionTemplate> t) {
 
 void StreamBase::RegisterExternalReferences(
     ExternalReferenceRegistry* registry) {
+  static bool is_registered = false;
```

Comment:
Is `RegisterExternalReferences` called by a single thread only?

Reply:
I believe so, yes, but we should definitely verify. If it is, it would still be good to add a comment here stating that fact so that it's clear to someone coming into this code later.

Reply:
@tniessen Yeah @jasnell, is correct.

The NodeMainInstance c'tor is run only during the the start up and it calls https://github.com/nodejs/node/blob/156365ebfc4493592f223be208f28a9e7f3ef851/src/node_main_instance.cc#L74-L75
which calls
https://github.com/nodejs/node/blob/156365ebfc4493592f223be208f28a9e7f3ef851/src/node_snapshotable.cc#L164 and that creates an instance of the ExternalReferenceRegistry class in https://github.com/nodejs/node/blob/156365ebfc4493592f223be208f28a9e7f3ef851/src/node_snapshotable.cc#L158 which calls the RegisterExternalReferences() functions inside the c'tor in https://github.com/nodejs/node/blob/156365ebfc4493592f223be208f28a9e7f3ef851/src/node_external_reference.cc#L18-L25.

I have added the comment, PTAL.

Reply:
Thanks for verifying!